### ISSUE TITLE
Mask image load error, which not convert to one dimension label image . 

### ISSUE THREAD

**Issue by xinxinxinxu** [open-mmlab/mmsegmentation#2471]
my experiment has to use a custom segmentation dataset. lt contains eights classeshowever, when it loads the mask image in RAM. it seems do not transform the pixel value to label value.
So the negative value of loss appeared.
I don't know why, when I used other datasets it was well.
Error report is:
![image](https://user-images.githubusercontent.com/51226104/211565234-fb30d23a-27b6-4682-b1a0-3e48d0b39366.png)

the pixel data value dim is [1,256,256,3]
when test the other datasets is [1,1024,768], which is right.

my dataset class define at below:
`@DATASETS.register_module()
class XXXXDataset(BaseSegDataset):
    """modified from Pascal VOC dataset.

    Args:
        split (str): Split txt file for Pascal VOC.
    """
    METAINFO = dict(
        classes=('cls0', 'cls1', 'cls2', 'cls3', 'cls4', 'cls5', 'cls6', 'cls7',),
        palette=[[0,255,100], [0,128,150], [255,128,50], [255,0,0], [255,255,0], [128,0,200],[128,128,50],[0,0,200]])

    def __init__(self,
                 ann_file,
                 img_suffix='.png',
                 seg_map_suffix='.png',
                 **kwargs) -> None:
        super().__init__(
            img_suffix=img_suffix,
            seg_map_suffix=seg_map_suffix,
            ann_file=ann_file,
            **kwargs)
        assert self.file_client.exists(
            self.data_prefix['img_path']) and osp.isfile(self.ann_file)`

my dataset config:
`dataset_type = 'XXXXDataset'
data_root = '/home/xx/KD_for_opt2sar/data'
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations'),
    dict(
        type='RandomResize',
        scale=(256, 256),
        ratio_range=(0.5, 2.0),
        keep_ratio=True),
    dict(type='RandomCrop', crop_size=(256, 256), cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(type='PackSegInputs')
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='Resize', scale=(256, 256), keep_ratio=True),
    dict(type='LoadAnnotations'),
    dict(type='PackSegInputs')
]

train_dataloader = dict(
    batch_size=2,
    num_workers=4,
    persistent_workers=True,
    sampler=dict(type='InfiniteSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_prefix=dict(
            img_path='sar_cropped', seg_map_path='lbl_cropped'),
        ann_file='train.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations'),
            dict(
                type='RandomResize',
                scale=(256, 256),
                ratio_range=(0.5, 2.0),
                keep_ratio=True),
            dict(type='RandomCrop', crop_size=(256, 256), cat_max_ratio=0.75),
            dict(type='RandomFlip', prob=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(type='PackSegInputs')
        ]))

val_dataloader = dict(
    batch_size=1,
    num_workers=1,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_prefix=dict(
            img_path='sar_cropped', seg_map_path='lbl_cropped'),
        ann_file='val.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', scale=(256, 256), keep_ratio=True),
            dict(type='LoadAnnotations'),
            dict(type='PackSegInputs')
        ]))
test_dataloader = dict(
    batch_size=1,
    num_workers=1,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_prefix=dict(
            img_path='sar_cropped', seg_map_path='lbl_cropped'),
        ann_file='test.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', scale=(256, 256), keep_ratio=True),
            dict(type='LoadAnnotations'),
            dict(type='PackSegInputs')
        ]))`

**Comment by MeowZheng** [comment]
As dataloader just loads the annotation but doesn't convert the annotation, I suggest you need to implement a script and transfer segmentation color image to segmentation mask before training or testing.
Here is an example 
https://github.com/open-mmlab/mmsegmentation/blob/9b8e8b730c2a17e9f0517489cb707e981e0eab39/tools/dataset_converters/vaihingen.py#L68-L80

**Comment by xiexinch** [comment]
Closing the issue, as there is no activity for a while.
We hope your issue has been resolved. 
If not, please feel free to open a new one.
